### ISSUE TITLE
Failed to open database '/data/data/com.my.packagename/databases/appboy.db'

### ISSUE THREAD

**Issue by yishai-glide** [Appboy/appboy-android-sdk#52]
Device: Samsung Galaxy Note 4
OS: 5.1.1
Library version: 1.13.3

this happens multiple times in our crash logs and I'm not sure what to do in order to fix this

```
06-07 11:51:23.235: E/SQLiteDatabase(32099): Failed to open database '/data/data/com.my.packagename/databases/appboy.db'.
06-07 11:51:23.235: E/SQLiteDatabase(32099): android.database.sqlite.SQLiteDatabaseLockedException: database is locked (code 5): , while compiling: PRAGMA journal_mode
06-07 11:51:23.235: E/SQLiteDatabase(32099): #################################################################
06-07 11:51:23.235: E/SQLiteDatabase(32099): Error Code : 5 (SQLITE_BUSY)
06-07 11:51:23.235: E/SQLiteDatabase(32099): Caused By : The database file is locked.
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	(database is locked (code 5): , while compiling: PRAGMA journal_mode)
06-07 11:51:23.235: E/SQLiteDatabase(32099): #################################################################
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1093)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.executeForString(SQLiteConnection.java:804)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.setJournalMode(SQLiteConnection.java:490)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.setWalModeFromConfiguration(SQLiteConnection.java:464)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:363)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnection.open(SQLiteConnection.java:228)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnectionPool.openConnectionLocked(SQLiteConnectionPool.java:512)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:206)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteConnectionPool.open(SQLiteConnectionPool.java:178)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteDatabase.openInner(SQLiteDatabase.java:908)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteDatabase.open(SQLiteDatabase.java:878)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:699)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.app.ContextImpl.openOrCreateDatabase(ContextImpl.java:1585)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.content.ContextWrapper.openOrCreateDatabase(ContextWrapper.java:283)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:223)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:163)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at bo.app.fa.c(SourceFile:48)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at bo.app.fa.b(SourceFile:166)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at bo.app.fa.a(SourceFile:61)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at bo.app.eq.run(SourceFile:31)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
06-07 11:51:23.235: E/SQLiteDatabase(32099): 	at java.lang.Thread.run(Thread.java:818)

```

**Comment by Bucimis** [comment]
Hi @yishai-glide,

Thanks for filing this issue.  We're aware of the issue and have a release planned for this week with a fix.

This issue comes from a race condition in our change-user code.  The race happens during an edge case in user cleanup and should have effectively no impact.

Please let us know if you have any comments/questions.  We'll be in contact again after the release.

Thanks,
Appboy Android Team

**Comment by yishai-glide** [comment]
I have many reports over the past few months of this if you require more information.

**Comment by Bucimis** [comment]
Hi @yishai-glide,

A fix for this bug was included in today's release version `1.13.5`.  If you could update to this release and confirm the issue is resolved it would be much appreciated.

Thanks,
Appboy Android Team

**Comment by briancaw** [comment]
Per confirmation of our fix working on https://github.com/Appboy/appboy-android-sdk/issues/19, which is an issue that stemmed from the same functionality in our code that caused this issue, I'm going to close this.  Please reopen if you find this has not been solved.

Thanks again for filing this issue,
Appboy Android Team
